src/subsystems/rust/translators/cargo-lock/default.nix

```diff
@@ -25,8 +25,23 @@ in {
     ...
   } @ args: let
     # get the root source and project source
-    rootSource = tree.fullPath;
-    projectSource = dlib.sanitizePath "${tree.fullPath}/${project.relPath}";
+    #
+    # SAFETY: We first let nix realize any context that might have not
+    # been realized yet. All of the string context should almost always
+    # be realized already, since sources pass through a discoverer, which
+    # travels through a source tree, so they must be realized. the only
+    # time this does not happen is if someone uses `translate` directly
+    # and passes an unrealized source.
+    #
+    # After this we can discard the string context and proceed as usual.
+    #
+    # If this is not done, when using the crane builder, the dependency
+    # derivation will capture the fullPath in it's `inputSrcs`, which causes
+    # frequent rebuilds (and the source is not needed anyways).
+    rootSource = l.unsafeDiscardStringContext (
+      l.seq (l.readFile "${tree.fullPath}/Cargo.toml") tree.fullPath
+    );
```

Comment:
I think it's better to use `builtins.path {path = "${tree.fullPath}/Cargo.toml", name = "Cargo.toml"}`.
This will make this path content addressed, like if it was coming from a FOD, and therefore will not cause rebuilds, despite the source changing.

Reply:
Okay, I made some changes to use `builtins.path` in some places and it seems to fix it for me.